src/models/user.model.ts

```diff
@@ -32,9 +33,28 @@ class User extends BaseModel implements Selectable<UserTable> {
     this.password = user.password
   }
 
-  isPasswordMatch = async (userPassword: string) => {
-    return bcrypt.compare(userPassword, this.password)
+  isPasswordMatch = async (userPassword: string): Promise<boolean> => {
+    if (!this.password) throw 'No password connected to user'
+    return await bcrypt.compare(userPassword, this.password || '')
+  }
+  static _convertArrayObjects(array: Selectable<UserTable>[]): User[] {
+    return array.reduce((arr: User[], obj: Selectable<UserTable>) => {
+      arr.push(this.convert(obj))
+      return arr
+    }, [])
   }
-}
 
-export { UserTable, User }
+  static convert<T extends Selectable<UserTable> | Selectable<UserTable>[]>(
```

Comment:
We can get the parent class at runtime, however we can't get the parent class at compilation time and thats when the types are compiled. So typescript can't get the type of the parent from the child.

However if it isn't a static function, then you can make the `BaseModel` generic and require a type argument, thats the type of the parent class, or the return type of convert. E.g.
```typescript
class User extends BaseModel<Selectable<UserTable>>
```
Then we could access the parent class type from the child class.

However, that creates some more problems, because then we would have to do something like `const user = new User().convert(...)`, and that's not great. However if you make the convert function the constructor, then it might be able to work, that would however override the constructor. But if you don't need a separate `constructor` and `convert` function then it might be able to work, because then you'd just use it like a normal instance `const user = new User(...)`.

Reply:
Yeah can you have a go at doing that then if we make it the constructor? It is probably better than the way I did it

Reply:
I'll give it a go

Reply:
Turns out we really didn't need to do anything fancy at all, I removed the convert methods and you just straightup don't need to implement them, cause the work for that is already done in the user constructor. So instead of doing `User.convert(...)` we now just do `new User(...)`.

The only caveat is we can't convert arrays like that, because a constructor can only return a singular type of the class, so instead of `const users = User.conver(userArray)` we just do `const users = userArray.map((user) => new User(user))`

And then everything is good

Reply:
Yeah the second part was the one I was stuck on

Reply:
Ahh the array part was why I originally did the convert way. This is a lot better.